### DISCUSSION TITLE
How to change value of d/qword using python

### DISCUSSION THREAD

**Discussion by commonuserlol** [NationalSecurityAgency/ghidra#6729] [Q&A]
Something like `put_dword` / `put_qword` in IDA?

**Comment by ryanmkurtz** [comment]
Should just be `setInt(address, value)` or `setLong(address, value)`.  

https://github.com/NationalSecurityAgency/ghidra/blob/0857d52681687835a20cf8a898e5c317ae5d632e/Ghidra/Features/Base/src/main/java/ghidra/program/flatapi/FlatProgramAPI.java#L2014-L2022

**Reply by commonuserlol** [reply]
I'm modifing `.got`, so it would point into my memory block. I found solution as:
```python
def write_address(addr, value):
	memory = currentProgram.getMemory()
	address = currentAddress.getAddress(addr)
	bytes = struct.pack("L" if currentAddress.getPointerSize() == 8 else "I", value)
	memory.setBytes(address, bytes)
```

`write_address(0xdeadbeef, 0x1337)` produces `00 00 13 37 00 00 00 00` in listing window
`memory.setInt(0xdeadbeef, 0x1337)` produces `37 13 00 00 00 00 00 00`

Which one is correct?
